sched/wqueue/kwork_cancel.c

```diff
@@ -93,6 +96,21 @@ static int work_qcancel(FAR struct kwork_wqueue_s *wqueue,
       work->worker = NULL;
       ret = OK;
     }
+  else if (nthread > 0)
+    {
+      int wndx;
+
+      for (wndx = 0; wndx < nthread; wndx++)
+        {
+          if (wqueue->worker[wndx].work == work &&
+              wqueue->worker[wndx].pid != _SCHED_GETTID())
```

Comment:
Why _SCHED_GETTID macro is used? It is not a libc call, so we should call nx call directly here

Reply:
since we want to avoid sched note syscall hook catch the kernel internal activities.

Reply:
Still not clear why we can't use `nxsched_gettid()` here. Could you please provide more detailed description?

Reply:
Please see this patch: https://github.com/apache/nuttx/pull/8394

Reply:
Yes, exactly. The `_SCHED_GETTID` is used in the places where code is shared between apps and kernel, like `libc`, so the question is do we compile code for kworker out of kernel?

Reply:
change to nx API